### PR TITLE
Return http 204 instead of 400 if no user

### LINKED ISSUE #656: The resend_activation endpoint discloses if a user with a given email exists
The `resend_activation` endpoint returns a 400 response if the given email does not belong to an (inactive) user.

This endpoint re-uses the password reset serializer (#555) but does not respect the `PASSWORD_RESET_SHOW_EMAIL_NOT_FOUND` setting because of these lines:

https://github.com/sunscrapers/djoser/blob/c62371e3f9a8bbad2eaf55ffd0efad6eb6c02f26/djoser/views.py#L208-L209

All settings related to disclosing email default to `False`:

```
PASSWORD_RESET_SHOW_EMAIL_NOT_FOUND
USERNAME_RESET_SHOW_EMAIL_NOT_FOUND
```

`resend_activation` shouldn't break this default.

P.S.

These settings don't work as advertised by the way, setting one to True has the effect of also toggling the other: 
https://github.com/sunscrapers/djoser/blob/c62371e3f9a8bbad2eaf55ffd0efad6eb6c02f26/djoser/serializers.py#L145-L149

### PR DESCRIPTION
**Pull request by jaredleishman**
Resolves #656 and matches what the [documentation](https://djoser.readthedocs.io/en/latest/settings.html#password-reset-show-email-not-found) says.
@tomwojcik are you the current maintainer? I saw your [comment](https://github.com/sunscrapers/djoser/issues/703#issuecomment-1493325630).

### CODE REVIEW

**Review by tomwojcik — APPROVED** [review]

**Comment by tomwojcik** [comment]
Yes, thanks for the PR. It will make it into the next release.

**Comment by tomwojcik** [comment]
Could you please rebase with master? I need to see if the CI is green.

**Comment by jaredleishman** [comment]
Should be rebased now @tomwojcik.

File: djoser/views.py [hunk 1]
```diff
@@ -202,7 +202,7 @@ def resend_activation(self, request, *args, **kwargs):
         user = serializer.get_user(is_active=False)
 
         if not settings.SEND_ACTIVATION_EMAIL or not user:
-            return Response(status=status.HTTP_400_BAD_REQUEST)
+            return Response(status=status.HTTP_204_NO_CONTENT)
```
Comment (jaap3) [inline djoser/views.py:205]: Maybe change this to a "positive check"? I.e.:

```python
if settings.SEND_ACTIVATION_EMAIL and user:
    context = {"user": user}
    to = [get_user_email(user)]
    settings.EMAIL.activation(self.request, context).send(to)
```

That way this method has only one point of return, this seems to be more inline with how the other methods are coded.

Comment (jaredleishman) [reply djoser/views.py]: Good idea. Updated.

**Comment by haxoza** [comment]
> Resolves #656 and matches what the [documentation](https://djoser.readthedocs.io/en/latest/settings.html#password-reset-show-email-not-found) says.

Just a heads-up - there is other piece of docs that would need to be updated in case of this change:

https://djoser.readthedocs.io/en/latest/base_endpoints.html#user-resend-activation-e-mail

**Comment by jaredleishman** [comment]
@tomwojcik and @jaap3 I've made some changes to introduce a new setting `RESEND_ACTIVATION_SHOW_EMAIL_NOT_FOUND` so the `/users/resend_activation/` endpoint behaves more like the `/users/reset_password/` endpoint.

File: djoser/serializers.py [hunk 2]
```diff
@@ -145,6 +145,7 @@ def get_user(self, is_active=True):
         if (
             settings.PASSWORD_RESET_SHOW_EMAIL_NOT_FOUND
             or settings.USERNAME_RESET_SHOW_EMAIL_NOT_FOUND
+            or settings.RESEND_ACTIVATION_SHOW_EMAIL_NOT_FOUND
```
Comment (jaap3) [inline djoser/serializers.py:148]: As noted in #656, I think this logic is flawed. Setting any one of these settings to `True` has the same effect as setting them all to `True`. These settings imply a granularity that is not actually achieved through code.

It's debatable if these settings should even be there. Account Enumeration is widely regarded as a security flaw, and these flows possibly disclosing the existence of an account are an attack vector.

(note, I'm not a maintainer of Djoser, this is just my opinion, I defer any decision about this to the actual maintainers.)

Comment (jaredleishman) [reply djoser/serializers.py:148]: Duly noted. I am happy to make changes to address more granularity or remove these settings altogether. Not sure how the maintainers want to move forward though.

Comment (tomwojcik) [reply djoser/serializers.py:148]: > As noted in #656, I think this logic is flawed. Setting any one of these settings to True has the same effect as setting them all to True. These settings imply a granularity that is not actually achieved through code.

I agree but to me it sounds like a follow up. A small refactor is somewhere on the roadmap, I need to get familiar with the codebase again.

> It's debatable if these settings should even be there. Account Enumeration is widely regarded as a security flaw, and these flows possibly disclosing the existence of an account are an attack vector.

Public tools have different requirements than internal tools. Defaults are safe so I see no harm being done by just having them. 

@jaredleishman would you be willing to just drop this new setting? Your changes with views will mitigate 400 if no user exists and I agree that adding `RESEND_ACTIVATION_SHOW_EMAIL_NOT_FOUND` might not be the best way forward.

Comment (tomwojcik) [reply djoser/serializers.py:148]: @jaredleishman bump

Comment (jaredleishman) [reply djoser/serializers.py]: Hey @tomwojcik, sorry I didn't see your response to this thread until now. I've removed that new setting.

File: testproject/testapp/tests/test_resend_activation.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 from testapp.tests.common import create_user, mock
 
 from djoser.compat import get_user_email
+from djoser.conf import settings as default_settings
```
Comment (tomwojcik) [inline testproject/testapp/tests/test_resend_activation.py:11]: CI fails because of this unused import. Please just remove it and then we're good to go.

**Review by tomwojcik — CHANGES_REQUESTED** [review]

File: djoser/serializers.py (hunk 2 above)
Comment (tomwojcik) [reply djoser/serializers.py]: No worries, thanks for pushing the requested changes. LGTM, just code quality step needs to be fixed.

File: testproject/testapp/tests/test_resend_activation.py (hunk 3 above)
Comment (jaredleishman) [reply testproject/testapp/tests/test_resend_activation.py]: Removed. Thanks @tomwojcik

**Comment by tomwojcik** [comment]
Thank you!

**Review by tomwojcik — APPROVED** [review]

**Pull request merged**